File: codes/greedy/455.assign-cookies/assign-cookies.cpp
#include <vector>
#include <algorithm>

using namespace std;
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(s.begin(), s.end());
        sort(g.begin(), g.end());
        auto sit = s.cbegin();
        auto git = g.cbegin();
        int sum = 0;
        while (sit != s.cend() && git != g.cend()) {
            while (*git > *sit) {
                if (++sit == s.cend()) {
                    return sum;
                }
            }
            sum++;
            sit++;
            git++;
        }
        return sum;
    }
};

File: codes/greedy/455.assign-cookies/assign-cookies.go
func findContentChildren(g []int, s []int) (sum int) {
	sort.Ints(s)
	sort.Ints(g)

	for i, j := 0, 0; i < len(g) && j < len(s); {
		for s[j] < g[i] {
			j++
			if j >= len(s) {
				return
			}
		}
		sum++
		i++
		j++
	}
	return
}
